avocado/plugins/beaker_result.py

```diff
@@ -0,0 +1,137 @@
+        baseurl = os.environ.get('BEAKER_LAB_CONTROLLER_URL')
+        recipeid = os.environ.get('RSTRNT_RECIPEID')
+        taskid = os.environ.get('RSTRNT_TASKID')
+        if baseurl is None or recipeid is None or taskid is None:
+            return
+        baseurl = baseurl.rstrip('/')
+        self.beaker_url = baseurl + '/recipes/' + recipeid + '/tasks/' + taskid
+        LOG_UI.info("beaker: using API at %s (R:%s T:%s)", baseurl, recipeid, taskid)
+
+    def send_request(self, req):
+        LOG_UI.info('beaker: %s %s ...', req.method, req.full_url)
+        try:
+            res = urllib.request.urlopen(req)  # nosec
+            return res
+        except urllib.error.URLError as err:
+            LOG_UI.info('beaker: Failed: %s', err)
+            return None
+        except Exception as err:
+            # should not happen
+            LOG_UI.info('beaker: Oops: %s', err)
+            return None
+
+    def post_result(self, state):
+        reqdict = {
+            'path': "%s/%s" % (self.job_id, state.get('name')),
+        }
+
+        result = state.get('status').lower().capitalize()
+        if result == 'Cancel':
+            result = 'Skip'
+        if result not in ['Pass', 'Fail', 'Skip']:
+            result = 'Fail'
+        reqdict['result'] = result
+
+        reason = state.get('fail_reason')
+        if reason is not None:
+            reqdict['message'] = reason
+
+        secs = state.get('time_elapsed')
+        if secs is not None:
+            reqdict['score'] = str(int(secs))
+
+        reqdata = urllib.parse.urlencode(reqdict).encode('utf-8')
+        url = self.beaker_url + '/results/'
+        req = urllib.request.Request(url, method='POST', data=reqdata)
+        res = self.send_request(req)
+        if res is None:
+            return None
+        return res.getheader('Location')
+
+    def put_data(self, location, name, content):
+        url = location + '/logs/' + name
+        req = urllib.request.Request(url, method='PUT', data=content)
+        self.send_request(req)
+
+    def put_file(self, location, name, filename):
+        file = open(filename)
+        content = file.read().encode('utf-8')
+        file.close()
+        self.put_data(location, name, content)
+
+    def put_file_list(self, location, prefix, filelist):
+        for file in filelist:
+            if os.path.isfile(file) and os.path.getsize(file) > 0:
+                name = prefix + os.path.basename(file)
+                self.put_file(location, name, file)
+
+    def pre_tests(self, job):
+        self.job_id = job.unique_id[:6]
+        return
+
+    def start_test(self, result, state):
+        return
+
+    def test_progress(self, progress=False):
+        return
+
+    def end_test(self, result, state):
+        if self.beaker_url is None:
+            return
+
+        location = self.post_result(state)
+        if location is None:
+            return
+
+        logfile = state.get('logfile')
```

Comment:
The file specified by `logfile` will, to the best of my knowledge, always be inside the `logdir` directory.  So, given all files in `logdir` are also sent, this ends up being a duplicate file.

Reply:
Nope, logfile is filtered from the logdir file list because of that.  The first logfile uploaded shows up more prominent in the beaker web ui, thats why the logfile gets the special treatment.

Reply:
Oh, I wasn't aware of that... never mind then.